plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/kafka/KafkaInfoPlugin.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.plugins.builtin.kafka
+
+import com.typesafe.config.Config
+import za.co.absa.atum.model.ControlMeasure
+import za.co.absa.enceladus.plugins.api.control.{ControlMetricsPlugin, ControlMetricsPluginFactory}
+
+/**
+ * Implementation of a plugin that sends control information to a Kafka topic when
+ * a checkpoint is created or when run status is changed.
+ */
+class KafkaInfoPlugin(producer: ControlInfoProducer) extends ControlMetricsPlugin {
```

Comment:
Sorry if I was not clear in my description. The name `KafkaInfoPlugin` suggests that it will have something to do with Kafka. But there is nothing that would connect it with Kafka in it's logic. Even the producer is `ControlInfoProducer` trait. That "Kafkaness" comes in the form that producer added in the factory is `ControlInfoProducerKafka`.
The discrepancy should be imho addressed, and I see two possibilities:
1) Rename the class and move it higher in the package hierarchy. E.g. `SenderInfoPlugin`. (My suggestion might be poor, feel free to use better)
2) Limit the producer to `ControlInfoProducerKafka`, bringing in the Kafka on constructor level.
Personally I think solution 1 is better.

Reply:
Yeah, the plugin itself has no dependency on Kafka. But the factory is dependent on Kafka. And I wanted to have the factory and the plugin to have the same name.